frontend/src/store/FamLoginUserState.ts

```diff
@@ -0,0 +1,166 @@
+export interface FamLoginUser {
+    username?: string;
+    displayName?: string;
+    email?: string;
+    idpProvider?: string; // from ID Token's ['identities']['providerName'] attribute.
+    roles?: string[]; // roles from Access Token's ['cognito:groups']. This may soon be redundant after delegated admin design.
+    authToken?: CognitoUserSession; // original JWT token from AWS Cognito (ID && Access Tokens).
+    accesses?: FamAuthGrantDto[]; // admin privileges retrieved from backend.
+};
+
+const state = ref({
+    famLoginUser: localStorage.getItem(FAM_LOGIN_USER)
+        ? (JSON.parse(localStorage.getItem(FAM_LOGIN_USER) as string) as
+              | FamLoginUser
+              | undefined
+              | null)
+        : undefined,
+});
+
+// --- getters
+
+const getAuthToken = () => {
+    return state.value.famLoginUser?.authToken;
+};
+
+const getUserAccess = () => {
+    return state.value.famLoginUser?.accesses;
+};
+
+// admin levels: 'FAM_ADMIN', 'APP_ADMIN', 'DELEGATED_ADMIN'
+const getUserAdminRoleGroups = () => {
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        return accesses.map(
+            (access) => access.auth_key
+        )
+    }
+};
+
+/**
+ * @returns array of applications for 'FAM_ADMIN' to be administered.
+ */
+const getAppsForFamAdminRole = (): FamApplicationDto[] | undefined => {
+    const accesses = getUserAccess();
+    if (accesses && accesses.length > 0) {
+        const access = accesses.filter(
+            (access) => access.auth_key == AdminRoleAuthGroup.FamAdmin
+        )[0]
+        return access
+            .grants
+            .map((grant) => grant.application)
+            .sort((first, second) => first.id - second.id);
+    }
+};
+
+/**
+ * Note!! this is to combine all applications the user has been granted
+ * and can administer on with different admin level privileges, only for
+ * special usage at dashboard page (ManagePermissions) since the user when
+ * it lands on the first page it needs to select "application" first.
+ *
+ * If the way our forntend works is based on admin level to display
+ * management table instead of selecting "applicatoin" first, it won't need
+ * this particular function.
+ */
+const getApplicationsUserAdministers = () => {
+    const accesses = getUserAccess();
+    let famApp;
+    const applicationList: Array<FamApplicationDto> = [];
+    if (accesses && accesses.length > 0) {
+        accesses.forEach((access) => {
+            const accessGrants = access.grants;
+            if (access.auth_key == AdminRoleAuthGroup.FamAdmin) {
+                famApp = accessGrants.filter(
+                    (grant) => grant.application.name == FAM_APPLICATION_NAME
+                )[0].application;
+            } else {
+                accessGrants.forEach((grant) => {
+                    const app = grant.application
+                    let isNewItem = true
+                    for (let item of applicationList) {
```

Comment:
Hi Ian, just wondering maybe we check if length of applicationList.filter((addedApp) => addedApp.id == app.id) > 0?  so we have less code? it doesn't impact functionality, I'll approve it, thanks!!

Reply:
Hi Catherine, I think it is fine, I don't always use short-form, sometimes I favor for-loop for clearness.
Thanks Catherine! I will merge so others can resolve their conflict.